core/src/main/java/ai/timefold/solver/core/impl/solver/termination/DiminishedReturnsTermination.java

```diff
@@ -22,7 +22,8 @@ final class DiminishedReturnsTermination<Solution_, Score_ extends Score<Score_>
     private final double minimumImprovementRatio;
 
     private boolean isGracePeriodActive;
-    private long gracePeriodStartTimeNanos;
+    boolean locked = true;
```

Comment:
I don't like that we're exposing internal synchronization details of the algorithm.

If you need to expose something for tests, please expose the current state - such as whether the period is active, or its start time.